server/conn.go

```diff
@@ -1507,17 +1507,17 @@ func (cc *clientConn) flush(ctx context.Context) error {
 
 func (cc *clientConn) writeOK(ctx context.Context) error {
 	msg := cc.ctx.LastMessage()
-	return cc.writeOkWith(ctx, msg, cc.ctx.AffectedRows(), cc.ctx.LastInsertID(), cc.ctx.Status(), cc.ctx.WarningCount())
+	return cc.writeOkWith(ctx, msg, cc.ctx.AffectedRows(), cc.ctx.LastInsertID(), cc.ctx.Status(), cc.ctx.WarningCount(), mysql.OKHeader)
 }
 
-func (cc *clientConn) writeOkWith(ctx context.Context, msg string, affectedRows, lastInsertID uint64, status, warnCnt uint16) error {
+func (cc *clientConn) writeOkWith(ctx context.Context, msg string, affectedRows, lastInsertID uint64, status, warnCnt uint16, header byte) error {
```

Comment:
Maybe rename to `writeWithHeader`?

Reply:
You mean `writeOkWithHeader` or `writeWithHeader`?

I think the current name is fine as it is always an `OK_Packet`, but with two kinds of headers (0x00, OK or 0xFE EOF). This is done in a weird way in the protocol, but that's just how it is.

See also:
- https://dev.mysql.com/doc/dev/mysql-server/latest/page_protocol_basic_ok_packet.html
- https://dev.mysql.com/doc/dev/mysql-server/latest/page_protocol_basic_eof_packet.html

I'd like to keep the current name if that's ok with you

Reply:
OK